kotlinx-coroutines-core/common/src/selects/Select.kt

```diff
@@ -621,9 +627,11 @@ internal open class SelectImplementation<R>(
                         // try to resume the continuation.
                         this.internalResult = internalResult
                         if (cont.tryResume(onCancellation)) return TRY_SELECT_SUCCESSFUL
-                        // If the resumption failed, we need to clean
-                        // the [result] field to avoid memory leaks.
-                        this.internalResult = null
+                        /*
+                         * If the resumption failed, we need to clean the [result] field to avoid memory leaks.
+                         * This write is benignly races with the very same write in cancellation invoke() handler
```

Comment:
And this one I don't like: to (ensure the correctness of)/(understand) this code, I don't think we need to know that there's a race.

A typo:
```suggestion
                         * This write benignly races with the very same write in cancellation invoke() handler
```